### PR TITLE
Add in system test code for uss issue command

### PR DESCRIPTION
**Pull request by ChrisAtkinson1**
Add in test code

### CODE REVIEW

**Comment by matthewpwilson** [comment]
@ChrisPark89 is feat/uss the main branch you're working off i.e. the one you'll submit the final PR to zowe/cli?

File: packages/zosuss/__tests__/__system__/cli/issue/cli.issue.ssh.with.cwd.system.test.ts [hunk 1]
```diff
@@ -0,0 +1,361 @@
...
+    });
+
+    it("should resolve --cwd option", async () => {
+        const commandName = "pwd";
+        const cwd =  `${defaultSystem.unix.testdir}/`;
+        Imperative.console.info("Resolve --cwd Command:" + commandName +"--cwd /" +cwd);
+        const response = await runCliScript(__dirname + "/__scripts__/issue_ssh_with_cwd.sh", TEST_ENVIRONMENT, [commandName, "/" + cwd]);
+
+        checkResponse(response);
+        // match only "/" with no following alpha-numeric character
```
Comment (matthewpwilson) [inline packages/zosuss/__tests__/__system__/cli/issue/cli.issue.ssh.with.cwd.system.test.ts]: Is this comment true?

File: packages/zosuss/__tests__/__system__/cli/issue/cli.issue.ssh.with.cwd.system.test.ts [hunk 2]
```diff
@@ -0,0 +1,361 @@
...
+    it("should get directory invalid --cwd option", async () => {
+        const zosmfProperties = TEST_ENVIRONMENT.systemTestProperties.systems.common.zosmf;
+
+        const commandName = "pwd";
+        const cwd = "/invaliddir";
+        Imperative.console.info("Invalid directory Command:" + commandName +"--cwd /" +cwd);
+        const response = await runCliScript(__dirname + "/__scripts__/issue_ssh_with_cwd.sh", TEST_ENVIRONMENT, [commandName, "/" + cwd]);
+
+        checkResponse(response);
+        // match only "/" with no following alpha-numeric character
```
Comment (matthewpwilson) [inline packages/zosuss/__tests__/__system__/cli/issue/cli.issue.ssh.with.cwd.system.test.ts]: I don't think this comment applies

File: packages/zosuss/__tests__/__system__/cli/issue/cli.issue.ssh.with.cwd.system.test.ts [hunk 3]
```diff
@@ -0,0 +1,361 @@
...
+
+        systemProps = new TestProperties(TEST_ENVIRONMENT.systemTestProperties);
+        defaultSystem = systemProps.getDefaultSystem();
+        const directory = `${defaultSystem.unix.testdir}/`;
+        // create a directory that the random dir will be created in
+        const commandName = "mkdir " + directory + "usscwdtest";
+        Imperative.console.info("Make test directory cmd:" + commandName);
+        const response = await runCliScript(__dirname + "/__scripts__/issue_ssh_no_cwd.sh", TEST_ENVIRONMENT, [commandName]);
+        checkResponse(response);
+        // match only "/" with no following alpha-numeric character
```
Comment (matthewpwilson) [inline packages/zosuss/__tests__/__system__/cli/issue/cli.issue.ssh.with.cwd.system.test.ts]: spurious comment again

File: packages/zosuss/__tests__/__system__/cli/issue/cli.issue.ssh.with.cwd.system.test.ts [hunk 4]
```diff
@@ -0,0 +1,361 @@
...
+        expect(response.stdout.toString()).toContain(randomDir);
+    });
+
+    it("should run install npm express", async () => {
+        const zosmfProperties = TEST_ENVIRONMENT.systemTestProperties.systems.common.zosmf;
+        const directory = `${defaultSystem.unix.testdir}/`;
+        const testdir = directory + "usscwdtest/";
+// tslint:disable-next-line: max-line-length
+        const commandName = "cd " + testdir + " &&npm install --registry http://cicsk8sm.hursley.ibm.com:30799 express";
```
Comment (matthewpwilson) [inline packages/zosuss/__tests__/__system__/cli/issue/cli.issue.ssh.with.cwd.system.test.ts]: I don't think we can use this registry in the tests as it's on the IBM network so the test won't work in the Zowe CI pipeline

File: packages/zosuss/__tests__/__system__/cli/issue/cli.issue.ssh.with.cwd.system.test.ts [hunk 5]
```diff
@@ -0,0 +1,361 @@
...
+        const cwd =  `${defaultSystem.unix.testdir}/`;
+        Imperative.console.info("Long Dir Command:" + commandName +"--cwd /" +cwd);
+        const response = await runCliScript(__dirname + "/__scripts__/issue_ssh_with_cwd.sh", TEST_ENVIRONMENT, [commandName, "/" + cwd]);
+
+        checkResponse(response);
+        // match only "/" with no following alpha-numeric character
+        expect(response.stdout.toString()).toContain(randomDir);
+    });
+
+    it("should run install npm express", async () => {
```
Comment (matthewpwilson) [inline packages/zosuss/__tests__/__system__/cli/issue/cli.issue.ssh.with.cwd.system.test.ts]: This test requires npm to be available on the target system... I'm not sure we can assume that

File: packages/zosuss/__tests__/__system__/cli/issue/cli.issue.ssh.with.cwd.system.test.ts (hunk 1 above)
Comment (ChrisPark89) [reply packages/zosuss/__tests__/__system__/cli/issue/cli.issue.ssh.with.cwd.system.test.ts]: I guess it should be "match only "/"+cwd with no following alpha-numeric character"

File: packages/zosuss/__tests__/__system__/cli/issue/cli.issue.ssh.without.cwd.system.test.ts [hunk 6]
```diff
@@ -0,0 +1,337 @@
...
+        TEST_ENVIRONMENT = await TestEnvironment.setUp({
+            testName: "issue_ssh",
+            tempProfileTypes: ["ssh","zosmf"]
+        });
+
+
+        systemProps = new TestProperties(TEST_ENVIRONMENT.systemTestProperties);
+        defaultSystem = systemProps.getDefaultSystem();
+
+        REAL_SESSION = TestEnvironment.createZosmfSession(TEST_ENVIRONMENT);
```
Comment (ChrisPark89) [inline packages/zosuss/__tests__/__system__/cli/issue/cli.issue.ssh.without.cwd.system.test.ts]: createZosmfSession need to be changed to createSshSession

File: packages/zosuss/__tests__/__system__/cli/issue/cli.issue.ssh.with.cwd.system.test.ts [hunk 7]
```diff
@@ -0,0 +1,362 @@
...
+        const commandName = "rm -rf " + directory + "usscwdtest";
+        Imperative.console.info("Remove test directory cmd:" + commandName);
+        const response = await runCliScript(__dirname + "/__scripts__/issue_ssh_no_cwd.sh", TEST_ENVIRONMENT, [commandName]);
+        Imperative.console.info("Remove Response:" + response.stdout.toString());
+        await TestEnvironment.cleanUp(TEST_ENVIRONMENT);
+
+    });
+
+    it.skip("should write a long directory", async () => {
+        const zosmfProperties = TEST_ENVIRONMENT.systemTestProperties.systems.common.zosmf;
```
Comment (ChrisPark89) [inline packages/zosuss/__tests__/__system__/cli/issue/cli.issue.ssh.with.cwd.system.test.ts]: zosmfProperties is not used in the test

File: packages/zosuss/__tests__/__system__/cli/issue/cli.issue.ssh.with.cwd.system.test.ts [hunk 8]
```diff
@@ -0,0 +1,362 @@
...
+        Imperative.console.info("Long Dir Command:" + commandName +"--cwd /" +cwd);
+        const response = await runCliScript(__dirname + "/__scripts__/issue_ssh_with_cwd.sh", TEST_ENVIRONMENT, [commandName, "/" + cwd]);
+
+        checkResponse(response);
+        // match only "/" with no following alpha-numeric character
+        expect(response.stdout.toString()).toContain(randomDir);
+    });
+
+    it("should run install npm express", async () => {
+        const zosmfProperties = TEST_ENVIRONMENT.systemTestProperties.systems.common.zosmf;
```
Comment (ChrisPark89) [inline packages/zosuss/__tests__/__system__/cli/issue/cli.issue.ssh.with.cwd.system.test.ts]: zosmfProperties is not used in the test

File: packages/zosuss/__tests__/__system__/cli/issue/cli.issue.ssh.with.cwd.system.test.ts [hunk 9]
```diff
@@ -0,0 +1,362 @@
...
+        Imperative.console.info("Run npm Command:" + commandName +"--cwd /" +cwd);
+        const response = await runCliScript(__dirname + "/__scripts__/issue_ssh_with_cwd.sh", TEST_ENVIRONMENT, [commandName, "/" + cwd]);
+
+        checkResponse(response);
+        // match only "/" with no following alpha-numeric character
+        expect(response.stdout.toString()).toContain("express@4.16.4");
+    });
+
+    it("should run npm on nonexistant module", async () => {
+        const zosmfProperties = TEST_ENVIRONMENT.systemTestProperties.systems.common.zosmf;
```
Comment (ChrisPark89) [inline packages/zosuss/__tests__/__system__/cli/issue/cli.issue.ssh.with.cwd.system.test.ts]: zosmfProperties is not used in the test

File: packages/zosuss/__tests__/__system__/cli/issue/cli.issue.ssh.with.cwd.system.test.ts [hunk 10]
```diff
@@ -0,0 +1,362 @@
...
+                Imperative.console.info("Deleted :" + ussname);
+            } catch (err) {
+                error = err;
+            }
+
+
+        });
+
+        it.skip("script issues exit64", async () => {
+            const zosmfProperties = TEST_ENVIRONMENT.systemTestProperties.systems.common.zosmf;
```
Comment (ChrisPark89) [inline packages/zosuss/__tests__/__system__/cli/issue/cli.issue.ssh.with.cwd.system.test.ts]: zosmfProperties is not used in the test

File: packages/zosuss/__tests__/__system__/cli/issue/cli.issue.ssh.with.cwd.system.test.ts [hunk 11]
```diff
@@ -0,0 +1,362 @@
...
+
+            checkResponse(response);
+            // match only "/" with no following alpha-numeric character
+            expect(response.stderr.toString()).toBe("");
+            expect(response.status).toBe(0);
+            expect(response.stdout.toString()).not.toContain("About to exit64");
+        });
+
+        it("script sleeps for 5 mins", async () => {
+            const zosmfProperties = TEST_ENVIRONMENT.systemTestProperties.systems.common.zosmf;
```
Comment (ChrisPark89) [inline packages/zosuss/__tests__/__system__/cli/issue/cli.issue.ssh.with.cwd.system.test.ts]: zosmfProperties is not used in the test

File: packages/zosuss/__tests__/__system__/cli/issue/cli.issue.ssh.with.cwd.system.test.ts [hunk 12]
```diff
@@ -0,0 +1,362 @@
...
+            checkResponse(response);
+            // match only "/" with no following alpha-numeric character
+            expect(response.stderr.toString()).toBe("");
+            expect(response.status).toBe(0);
+            expect(response.stdout.toString()).toContain("FINISHED");
+        });
+
+
+        it("script asks for input", async () => {
+            const zosmfProperties = TEST_ENVIRONMENT.systemTestProperties.systems.common.zosmf;
```
Comment (ChrisPark89) [inline packages/zosuss/__tests__/__system__/cli/issue/cli.issue.ssh.with.cwd.system.test.ts]: zosmfProperties is not used in the test

File: packages/zosuss/__tests__/__system__/cli/issue/cli.issue.ssh.with.cwd.system.test.ts [hunk 13]
```diff
@@ -0,0 +1,362 @@
...
+            const response = await runCliScript(__dirname + "/__scripts__/issue_ssh_with_cwd.sh", TEST_ENVIRONMENT, [commandName,"/" + cwd]);
+
+            checkResponse(response);
+            expect(response.stderr.toString()).toBe("");
+            expect(response.status).toBe(0);
+            expect(response.stdout.toString()).toContain("Welcome Tester!");
+        });
+
+        it("script kills itself", async () => {
+            const zosmfProperties = TEST_ENVIRONMENT.systemTestProperties.systems.common.zosmf;
```
Comment (ChrisPark89) [inline packages/zosuss/__tests__/__system__/cli/issue/cli.issue.ssh.with.cwd.system.test.ts]: zosmfProperties is not used in the test

**Comment by ChrisPark89** [comment]
The test took 986 seconds (roughly 16 minutes) to run. Also, while it is running, it deson't give any output. We may need to do something about it.

File: packages/zosuss/__tests__/__system__/cli/issue/cli.issue.ssh.with.cwd.system.test.ts (hunk 5 above)
Comment (ChrisPark89) [reply packages/zosuss/__tests__/__system__/cli/issue/cli.issue.ssh.with.cwd.system.test.ts]: I agree with Matthew.

File: packages/zosuss/__tests__/__system__/cli/issue/cli.issue.ssh.system.test.ts [hunk 14]
```diff
@@ -46,11 +70,228 @@ describe("zowe uss issue ssh", () => {
...
+        const commandName = "cd " + directory + " && chmod 777 exit64.sh && exit64.sh";
+        // Imperative.console.info("Exit command:" + commandName);
+        const response = await runCliScript(__dirname + "/__scripts__/issue_ssh_no_cwd.sh", TEST_ENVIRONMENT, [commandName]);
+
+        checkResponse(response);
+        expect(response.stdout.toString()).not.toContain("About to exit64");
+    });
+
+    /*
+    it("script sleeps for 5 mins", async () => {
```
Comment (ChrisPark89) [inline packages/zosuss/__tests__/__system__/cli/issue/cli.issue.ssh.system.test.ts]: Do we need this 5 minutes sleep test? This slows down running the test significantly.

Comment (matthewpwilson) [reply packages/zosuss/__tests__/__system__/cli/issue/cli.issue.ssh.system.test.ts]: Yes I agree this one shouldn't be path of the automated system test

Comment (ChrisPark89) [reply packages/zosuss/__tests__/__system__/cli/issue/cli.issue.ssh.system.test.ts]: It's commented out for now

**Pull request merged**